server/game/player.js

```diff
@@ -628,7 +628,11 @@ class Player extends Spectator {
     recyclePlots() {
         if(this.plotDeck.isEmpty()) {
             this.plotDiscard.each(plot => {
-                this.moveCard(plot, 'plot deck');
+                if(this.agenda && this.agenda.code === '05045' && plot.hasTrait('Scheme')) {
```

Comment:
It would be better if the Player class had no knowledge whatsoever of the rules for the Rains agenda. Is there an alternate way for this to happen on the RainsOfCastamere class instead?

Perhaps if Player moves them all to the plot deck per normal rules, raises an event after the plots have been recycled, which Rains then listens for and then re-separates the Scheme pile (like it does onDeckPrepared at the start of the game)? You get a double move which could be a problem for future cards, but it wouldn't be a problem currently.

You might also be able to listen to the existing onCardMoved event for Schemes being moved to the plot deck and move it again. Same problem of double move but wouldn't require introducing a new event. Though additional care may be required there - that event is fired on every card move, it may interfere with the existing onDecksPrepared (might make it no longer needed, might interact unpredictably), etc.